### PR TITLE
Implement functionality to determine maximum number of items shown in the relation editor widget

### PR DESCRIPTION
**Pull request by nirvn**
See https://github.com/opengisch/QField/pull/3520 for more details on functionality.

### CODE REVIEW

File: qfieldsync/gui/map_layer_config_widget.py [hunk 1]
```diff
@@ -70,15 +73,31 @@ def __init__(self, layer, canvas, parent):
...
-            self.layout().insertRow(2, self.tr("Photo naming"), self.photoNamingTable)
+            self.layout().insertRow(-1, self.tr("Photo naming"), self.photoNamingTable)
             self.photoNamingTable.setEnabled(self.photoNamingTable.rowCount() > 0)
 
+        self.relationshipConfigurationTable = RelationshipConfigurationTableWidget()
+        self.relationshipConfigurationTable.addLayerFields(self.layer_source)
+        self.relationshipConfigurationTable.setLayerColumnHidden(True)
+
+        # append the relationship configuration table to the layout
+        if Qgis.QGIS_VERSION_INT >= 31300 and layer.type() == QgsMapLayer.VectorLayer:
```
Comment (suricactus) [inline qfieldsync/gui/map_layer_config_widget.py:91]: I think we can forget about QGIS 3.14 and simplify the check here and on line 82.

File: qfieldsync/gui/map_layer_config_widget.py [hunk 2]
```diff
@@ -28,6 +28,9 @@
 
 from qfieldsync.core.message_bus import message_bus
 from qfieldsync.gui.photo_naming_widget import PhotoNamingTableWidget
+from qfieldsync.gui.relationship_configuration_widget import (
+    RelationshipConfigurationTableWidget,
```
Comment (suricactus) [inline qfieldsync/gui/map_layer_config_widget.py:32]: @nirvn hm, forgotten to commit?

Comment (nirvn) [reply qfieldsync/gui/map_layer_config_widget.py:32]: @suricactus , oups, added.

File: qfieldsync/gui/relationship_configuration_widget.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+        if layer.type() != QgsMapLayer.VectorLayer:
+            return
+
+        for relation in (
+            QgsProject.instance().relationManager().referencedRelations(layer)
+        ):
+            row = self.rowCount()
+            self.insertRow(row)
+            item = QTableWidgetItem(layer.name())
```
Comment (suricactus) [inline qfieldsync/gui/relationship_configuration_widget.py:59]: can we have proper names for these `item`s, like `layer_name_item`, `relation_id_item` etc?

File: qfieldsync/gui/relationship_configuration_widget.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            item.setFlags(Qt.ItemIsEnabled)
+            self.setItem(row, 1, item)
+            item = QTableWidgetItem(relation.name())
+            item.setFlags(Qt.ItemIsEnabled)
+            self.setItem(row, 2, item)
+            spin = QgsSpinBox()
+            spin.setMinimum(1)
+            spin.setMaximum(100)
+            spin.setSingleStep(1)
+            spin.setValue(layer_source.relationship_maximum_visible(relation.id()))
```
Comment (suricactus) [inline qfieldsync/gui/relationship_configuration_widget.py:73]: Would be great if we can set it to `0` or `NULL` and for QFiield this means: grow the relationship widget to follow the number of related items.

**Review by suricactus — APPROVED** [review]
Looks good to me. Small functionality suggestion that would improve life a lot and one variable naming issue.

**Comment by nirvn** [comment]
@suricactus , comments addressed, unlimited mode in.

**Review by suricactus — APPROVED** [review]

**Pull request merged**